crates/cheatnet/src/forking/state.rs

```diff
@@ -0,0 +1,73 @@
+use crate::forking::worker::Worker;
+use crate::state::DictStateReader;
+use blockifier::execution::contract_class::ContractClass;
+use blockifier::state::state_api::{StateReader, StateResult};
+use starknet_api::core::{ClassHash, CompiledClassHash, ContractAddress, Nonce};
+use starknet_api::hash::StarkFelt;
+use starknet_api::state::StorageKey;
+
+#[derive(Debug)]
+pub struct ForkStateReader {
+    pub dict_state_reader: DictStateReader,
+    pub worker: Worker,
+}
+
+impl StateReader for ForkStateReader {
+    fn get_storage_at(
```

Comment:
I think we should store the values in dictionaries (a simple cache without storing anything on disk)

Reply:
I think we're reading right now from the dictionary, but we're not writing to it on response before returning the value

Reply:
Or does it cache automatically after returning? I might be wrong here

Reply:
Yup, it is cached automatically in the CachedState which calls the `CustomStateReader`

Reply:
Unresolving, since this looks like a major concern to me
cc @MaksymilianDemitraszek

Reply:
> low prob that someone will want to use contract deployed at address e.g. 0x101

@piotmag769 what do you mean by that?